facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java

```diff
@@ -48,4 +67,65 @@ public BuildProviderImpl(BuildRecordRepository repository, BuildMapper mapper, B
 
         return queryForCollection(pageIndex, pageSize, sortingRsql, query, withPerformedInMilestone(milestoneId));
     }
+
+    @Override
+    public Page<Build> getBuildsForProject(int pageIndex,
+                                           int pageSize,
+                                           String sortingRsql,
+                                           String query,
+                                           Integer projectId) {
+
+        List<Object[]> buildConfigurationRevisions = AuditReaderFactory
+                .get(entityManager)
+                .createQuery()
+                .forRevisionsOfEntity(BuildConfiguration.class, false, false)
+                .add(AuditEntity.relatedId("project").eq(projectId))
```

Comment:
Relationship to project is not audited so this whole thing is pointless and wastes resources. Instead just do normal query with `queryForCollection`.

Reply:
Replaced with:

```
        @SuppressWarnings("unchecked")
        Set<Integer> buildConfigIds = buildConfigurationRepository
                .queryWithPredicates(withProjectId(projectId))
                .stream()
                .map(BuildConfiguration::getId)
                .collect(Collectors.toSet());

        return queryForCollection(pageIndex, pageSize, sortingRsql, query, withBuildConfigurationIds(buildConfigIds));
```

Reply:
BuildRecord has no projectId field, but has a relationship to project via it's associated BuildConfiguration